#! /usr/bin/env python

PACKAGE='controllers'

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       	Name		Type      	Level	Description				Default 	Min	Max
gen.add( 	"k_p_xy", 	double_t, 	0, 	"Proportional gain for x and y", 	2.0, 		0, 	10)  		# Slow: 2.0,	Fast: 4.0, 	VI: 2.0
gen.add( 	"k_i_xy",	double_t, 	0, 	"Integral gain for x and y", 		0.1, 		0, 	10) 		# Slow: 0.1,	Fast: 0.2, 	VI: 0.1
gen.add( 	"k_d_xy",	double_t, 	0, 	"Derivative gain for x and y", 		0.6, 		0, 	10) 		# Slow: 0.6,	Fast: 1.0, 	VI: 1.0
gen.add( 	"k_p_z", 	double_t, 	0, 	"Proportional gain for z", 		2.0, 		0, 	10)  		# Slow: 2.0,	Fast: 4.0, 	VI: 2.0
gen.add( 	"k_i_z",	double_t, 	0, 	"Integral gain for z", 			0.1, 		0, 	10) 		# Slow: 0.1,	Fast: 0.2, 	VI: 0.1
gen.add( 	"k_d_z",	double_t, 	0, 	"Derivative gain for z", 		0.6, 		0, 	10) 		# Slow: 0.6,	Fast: 1.0, 	VI: 0.1
gen.add( 	"k_p_yaw", 	double_t, 	0, 	"Proportional gain for yaw", 		0.2, 		0, 	10)  		# Slow: 2.0,	Fast: 4.0, 	VI: 0.2

exit(gen.generate(PACKAGE, "controllers", "setPID"))
